File: core/src/main/java/com/graphhopper/storage/CHConfig.java
package com.graphhopper.storage;

import com.graphhopper.config.Profile;
import com.graphhopper.routing.util.TraversalMode;
import com.graphhopper.routing.weighting.Weighting;

/**
 * Container to hold properties used for CH preparation Specifies all properties of a CH routing profile.
 *
 * @author easbar
 */
public class CHConfig {
    /**
     * will be used to store and identify the CH graph data on disk
     */
    private final String chGraphName;
    private final Weighting weighting;
    private final boolean edgeBased;

    public static CHConfig nodeBased(String chGraphName, Weighting weighting) {
        return new CHConfig(chGraphName, weighting, false);
    }

    public static CHConfig edgeBased(String chGraphName, Weighting weighting) {
        return new CHConfig(chGraphName, weighting, true);
    }

    public CHConfig(String chGraphName, Weighting weighting, boolean edgeBased) {
        Profile.validateProfileName(chGraphName);
        this.chGraphName = chGraphName;
        this.weighting = weighting;
        this.edgeBased = edgeBased;
    }

    public Weighting getWeighting() {
        return weighting;
    }

    public boolean isEdgeBased() {
        return edgeBased;
    }

    public TraversalMode getTraversalMode() {
        return edgeBased ? TraversalMode.EDGE_BASED : TraversalMode.NODE_BASED;
    }

    public String toFileName() {
        return chGraphName;
    }

    public String toString() {
        return chGraphName;
    }

    public String getName() {
        return chGraphName;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        CHConfig chConfig = (CHConfig) o;
        return getName().equals(chConfig.getName());
    }

    @Override
    public int hashCode() {
        return getName().hashCode();
    }
}

File: core/src/main/kotlin/com/graphhopper/storage/CHConfig.kt
package com.graphhopper.storage

import com.graphhopper.config.Profile
import com.graphhopper.routing.util.TraversalMode
import com.graphhopper.routing.weighting.Weighting

/**
 * Container to hold properties used for CH preparation Specifies all properties of a CH routing profile.
 *
 * @author easbar
 */
class CHConfig(chGraphName: String, val weighting: Weighting, val isEdgeBased: Boolean) {
    /**
     * will be used to store and identify the CH graph data on disk
     */
    private val chGraphName: String

    init {
        Profile.validateProfileName(chGraphName)
        this.chGraphName = chGraphName
    }

    val traversalMode: TraversalMode
        get() = if (isEdgeBased) TraversalMode.EDGE_BASED else TraversalMode.NODE_BASED

    fun toFileName(): String = chGraphName

    override fun toString(): String = chGraphName

    val name: String
        get() = chGraphName

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val chConfig = other as CHConfig
        return name == chConfig.name
    }

    override fun hashCode(): Int = name.hashCode()

    companion object {
        @JvmStatic
        fun nodeBased(chGraphName: String, weighting: Weighting): CHConfig =
            CHConfig(chGraphName, weighting, false)

        @JvmStatic
        fun edgeBased(chGraphName: String, weighting: Weighting): CHConfig =
            CHConfig(chGraphName, weighting, true)
    }
}
